src/Shared/CommunicationsUtilities.cs

```diff
@@ -665,12 +665,78 @@ internal static int GetIntegerVariableOrDefault(string environmentVariable, int
             return defaultValue;
         }
 
+        /// <summary>
+        /// Writes trace information to a log file
+        /// </summary>
+        internal static void Trace<T>(string format, T arg0)
+        {
```

Comment:
The JIT should be smart enough to inline this if it were just `Trace(nodeId: -1, format, arg0)`

Reply:
Yes, this can remove some lines. Thanks for suggestion. Originally, I tried to move ```nodeId``` and make it optional, but overload with ```params object[]``` was preferred by compiler in cases where I wanted to invoke different method :)